crates/wiggle/src/lib.rs

```diff
@@ -542,6 +542,26 @@ impl<'a, T> GuestPtr<'a, [T]> {
     pub fn as_ptr(&self) -> GuestPtr<'a, T> {
         GuestPtr::new(self.mem, self.offset_base())
     }
+
+    pub fn get(&self, index: u32) -> Option<GuestPtr<'a, T>> {
+        if index < self.len() {
+            Some(GuestPtr::new(self.mem, self.offset_base() + index))
```

Comment:
Should this be `index` times the gues size of `T`?

Reply:
len() is in `units` (correcting this to `elements` in the doc strings, not bytes, so I dont think we need to multiply by the size.

Reply:
Yeah the length is in units but the `GuestPtr` returned here I think is offset in bytes from the memory base?

Reply:
Thanks, I completely missed that. Clearly I'm not smart enough to write or maintain this code without tests, and since the only ones i have currently are in wasi-common and use `GuestPtr<[u8]>`, ill add some others to the wiggle tree that use non-byte-sized elements and make sure its done right.